tests/core/test_schema_components.py

```diff
@@ -53,6 +56,35 @@ def test_column_coerce() -> None:
     assert Engine.dtype(validated.a.dtype) == Engine.dtype(int)
 
 
+def test_config_coerce() -> None:
```

Comment:
One recommendation I have is to convert these into `DataFrameSchema`s instead of using `DataFrameModel` because the latter is essentially converted into the former when validation happens, i.e. `DataFrameSchema` the "true" representation of a schema.

They can live in `test_schemas.py` if you decide to change this. Also happy to accept these tests using `DataFrameModel`, in which case they ought to be in `test_model.py`